iron-autogrow-textarea.html

```diff
@@ -334,10 +334,8 @@
       this.bindValue = value;
       this.$.mirror.innerHTML = this._valueForMirror();
 
-      // This code is from early 1.0, when this element was a type extension.
-      // It's unclear if it's still needed, but leaving in in case it is.
-      // manually notify because we don't want to notify until after setting value.
-      // this.fire('bind-value-changed', {value: this.bindValue});
+      // Manually notify because we don't want to notify until after setting value.
+      this.fire('bind-value-changed', {value: this.bindValue});
```

Comment:
yes and no. The `bind-value-changed` is fired on `this.bindValue = value`, but not on a change from its parent:
```
<iron-autogrow-textarea bind-value="{{myvalue}}">
[...]
myvalue = "foo"
```
for some reason, `<paper-input-container>` relies on this and does not behave properly if `bind-value-changed` is not fired on EVERY (parent AND child) change.

I don't know why this was implemented like this, maybe because `<iron-input>` behaves the same way. (digging deeper into the code also revealed that you deprecated `bindValue`, while iron-input deprecates `value`, which is kind of inconsistent, but whatever...)

Now I totally agree that it is kind of a bad design to manually fire `foo-change` on a `foo` observer (which my patch does). But for backwards compatibility this is probably needed, unless you remove `notify: true` on the `value` property, which would be kind of sad (but again would be consistent with "iron-input").

I don't know how much backwards-compatibility you are targeting at polymer/google, maybe the data flow in `<paper-input>` should be reconsidered, probably by adding a dynamic observer instead of listening to events.